chain/client/src/stateless_validation/state_witness_tracker.rs

```diff
@@ -0,0 +1,213 @@
+use std::hash::{Hash, Hasher};
+use borsh::{BorshDeserialize, BorshSerialize};
+use lru::LruCache;
+use once_cell::sync::Lazy;
+
+use near_async::time::Clock;
+use near_o11y::metrics::{exponential_buckets, HistogramVec, try_create_histogram_vec};
+use near_primitives::sharding::ChunkHash;
+use near_primitives::stateless_validation::{ChunkStateWitness, ChunkStateWitnessAck};
+
+/// This is currently used for metrics computation, so we do not need to keep all the
+/// witnesses, so make the capacity 10.
+const CHUNK_STATE_WITNESS_MAX_RECORD_COUNT: usize = 5;
+
+static CHUNK_STATE_WITNESS_NETWORK_ROUNDTRIP_TIME: Lazy<HistogramVec> = Lazy::new(|| {
```

Comment:
nit: for some reason we have all our metrics in a separate file `chain/client/src/metrics.rs`, so maybe consider moving it there to keep it consistent

Reply:
I first added it there but then decided to move here to make the metric and the code closer. I do not have a specific preference, I can also move it there.